Revit_Core_Engine/Convert/Environment/ToRevit/Pipe.cs

```diff
@@ -0,0 +1,136 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.DB.Mechanical;
+using BH.Engine.Adapters.Revit;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Spatial.ShapeProfiles;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.MEP.System.SectionProperties;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        public static Autodesk.Revit.DB.Plumbing.Pipe ToRevitPipe(this oM.MEP.System.Pipe pipe, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            // Check valid pipe object
+            if (pipe == null)
+                return null;
+
+            // Construct Revit Pipe
+            Autodesk.Revit.DB.Plumbing.Pipe revitPipe = refObjects.GetValue<Autodesk.Revit.DB.Plumbing.Pipe>(document, pipe.BHoM_Guid);
+            if (revitPipe != null)
+                return revitPipe;
+
+            // Settings
+            settings = settings.DefaultIfNull();
+
+            // Pipe type
+            Autodesk.Revit.DB.Plumbing.PipeType pipeType = pipe.SectionProperty.ToRevitElementType(document, new List<BuiltInCategory> { BuiltInCategory.OST_PipingSystem }, settings, refObjects) as Autodesk.Revit.DB.Plumbing.PipeType;
+
+            // End points
+            XYZ start = pipe.StartPoint.ToRevit();
+            XYZ end = pipe.EndPoint.ToRevit();
+
+            // Level
+            Level level = document.LevelBelow(Math.Min(start.Z, end.Z), settings);
+            if (level == null)
+                return null;
+
+            // Default system used for now
+            // TODO: in the future you could look for the existing connectors and check if any of them overlaps with start/end of this pipe - if so, use it in Pipe.Create.
+            // hacky/heavy way of getting all connectors in the link below - however, i would rather filter the connecting elements out by type/bounding box first for performance reasons
+            // https://thebuildingcoder.typepad.com/blog/2010/06/retrieve-mep-elements-and-connectors.html
+
+            Autodesk.Revit.DB.Plumbing.PipingSystemType pst = new FilteredElementCollector(document).OfClass(typeof(Autodesk.Revit.DB.Plumbing.PipingSystemType)).OfType<Autodesk.Revit.DB.Plumbing.PipingSystemType>().FirstOrDefault();
```

Comment:
Maybe we could drop a warning that 1st available system is used? same refers to ducts/cabletrays etc.

Reply:
A warning message has been reflected across all objects within the PR scope 😄